src/kernel/chainparamsbase.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_KERNEL_CHAINPARAMSBASE_H
+#define BITCOIN_KERNEL_CHAINPARAMSBASE_H
+
+#include <string>
+
+/**
+ * CBaseChainParams defines the base parameters (shared between bitcoin-cli and bitcoind)
+ * of a given instance of the Bitcoin system.
+ */
+class CBaseChainParams
```

Comment:
In commit "split non/kernel chainparamsbase" (341be7fa680e9dcdbaa46db0194833b5d0229394)

I still don't think it makes sense to have `CBaseChainParams` in the kernel. Like the comment above says, the point of the `CBaseChainParams` class and the reason it is separate from `CChainParams` is that it is shared between bitcoin-cli and bitcoind. I think of `CBaseChainParams` as client parameters, and `CChainParams` as server parameters. Practically speaking, `bitcoin-cli` or other clients that needs basic information like network datadir locations or RPC port numbers should not be required to link against the kernel.

The justification for this change given in the commit message about allowing "kernel chainparams to be instantiated" also does not make sense to me, because chainparamsbase is part of libbitcoin_util, and it is fine for kernel code to depend on libbitcoin_util if we are going by https://github.com/bitcoin/bitcoin/blob/master/doc/design/libraries.md#dependencies

Previous comment about this was: https://github.com/bitcoin/bitcoin/pull/26177#discussion_r1126765676

Reply:
This makes sense, I wasn't connecting the fact that the util library includes `chainparamsbase` anyway with segregating the kernel code. I dropped the commit.